src-ts/proposalMonitor.ts

```diff
@@ -149,3 +131,68 @@ export class GovernorProposalMonitor extends EventEmitter {
     this.polling = false;
   }
 }
+
+/**
+ * Monitors a governor contract for created proposals. Starts a new proposal pipeline when
+ * a proposal is created.
+ */
+export class GovernorProposalMonitor extends ProposalMonitor {
+  public override async getOriginReceipts(
+    fromBlock: BlockTag,
+    toBlock: BlockTag,
+    proposalId?: string
+  ) {
+    const governor = L2ArbitrumGovernor__factory.connect(this.originAddress, this.originProvider);
+    const logs = await this.originProvider.getLogs({
+      fromBlock,
+      toBlock,
+      ...governor.filters.ProposalCreated(),
+    });
+
+    const filteredTxHashes = logs
+      .filter(
+        (l) =>
+          !proposalId ||
+          (
+            governor.interface.parseLog(l).args as unknown as ProposalCreatedEventObject
+          ).proposalId.toHexString() === proposalId
+      )
+      .map((l) => l.transactionHash);
+
+    return await Promise.all(
+      Array.from(new Set(filteredTxHashes)).map((t) => this.originProvider.getTransactionReceipt(t))
+    );
+  }
+}
+
+export class GnosisSafeProposalMonitor extends ProposalMonitor {
+  public override async getOriginReceipts(fromBlock: BlockTag, toBlock: BlockTag, txHash?: string) {
+    const gnosisSafeInterface = new Interface([
+      "event ExecutionSuccess(bytes32 indexed txHash, uint256 payment)",
+    ]);
+    const executionSuccessEvent = gnosisSafeInterface.encodeFilterTopics("ExecutionSuccess", []);
+    const logs = await this.originProvider.getLogs({
```

Comment:
does this handle gov-chain 9-of-12 direct calls the SC manager? (add, remove, rotate, etc.) (in additional to 7-of-12 proposals)

Reply:
Yeh, any call through a gnosis safe can be picked up here